js/data.js

```diff
@@ -1,8 +1,26 @@
 /* exported data */
+var previousEntries = null;
 
 var data = {
   view: 'entry-form',
   entries: [],
   editing: null,
   nextEntryId: 1
 };
+
+function loadData(event) {
+  previousEntries = localStorage.getItem('entries-local-storage');
+  if (previousEntries !== null) {
+    data.entries = JSON.parse(previousEntries).entries;
+    data.nextEntryId = JSON.parse(previousEntries).nextEntryId++;
+  }
+}
+
+function saveDataB4Unload(event) {
+  if (data.entries.length > 0) {
+    localStorage.setItem('entries-local-storage', JSON.stringify(data));
+  }
+}
+
+window.addEventListener('load', loadData);
```

Comment:
you don't need this here, you can just check local storage when the file is read.

Reply:
is there a reason why we would want to code to run when the script is loaded vs when the page is loaded?